src/targets/maven.ts

```diff
@@ -284,7 +294,9 @@ export class MavenTarget extends BaseTarget {
       pomFile,
     } = this.getFilesForMavenCli(distDir);
 
-    this.retrySpawnProcess(
+    // Maven central is very flaky, so retrying with an exponential delay in
+    // in case it fails.
+    retrySpawnProcess(
```

Comment:
This is not the API I was trying to convey, the function signature should be same as `spawnProcess`:
```js
await retrySpawnProcess(
  this.mavenConfig.mavenCliPath,
  [
    'gpg:sign-and-deploy-file',
    `-Durl=${this.mavenConfig.mavenRepoUrl}`,
    `--settings ${this.mavenConfig.mavenSettingsPath}`,
  ],
  {
    maxRetries: MAX_PUBLISHING_ATTEMPTS,
  }
);
```

Note that you are missing an `await` in front of this function, which will cause all kinds of confusing issues.

The `RETRY_DELAY_SECS` and `RETRY_EXP_FACTOR` constants should live in the `utils/system` module and be the defaults for this helper.

Reply:
Moved to `utils/async` instead, since mocks in `utils/system` were interfering, and IMO `async` was the most appropriate module to move to.

Reply:
Eh, okay 😀